### ISSUE TITLE
Any idea why FastText(en-news) beats XLNet(large-cased) in TREC6?

### ISSUE THREAD

**Issue by krzysztoffiok** [flairNLP/flair#1233]
I'm preparing a benchmark of selected LM's available in Flair in sentence classification task on TREC6. To my surprise 10-fold-cross validated results of training a sentence-classifier with exactly the same set of parameters and change only in the Language Model demonstrate that a framework based on FastText(en-news) beats XLNet(xlnet-large-cased). Any ideas for reasons why can that be (i evaluate based on Matthews Correlation Coefficient computed on test set)? I have evaluated also Elmo, Roberta, Glove, Flair, BytePairEmbeddings and these results make sense i.e. Roberta significantly wins in term of quality of prediction. Why doesn't it work "as expected" with XLNet? Can the "cased"/"uncased" flavor of model make such a big difference? Or maybe some of the model parameters that "work fine" for FastText  should be altered to make XLNet work "correctly"? I also get similarly poor and surprising results with GPT1.

![fasttext_xlnetcased](https://user-images.githubusercontent.com/37101271/67195119-ad831a00-f3f8-11e9-8b49-56a28b48706b.png)

i'm using ubuntu 18.04 LTS, flair 0.4.3. and following parameters for model and training
hidden_size=256,
reproject_words=True,
reproject_words_dimension=256,
rnn_type="gru",
bidirectional=False,
rnn_layers=1

max_epochs=1000,
learning_rate=0.1,
mini_batch_size=8,
anneal_factor=0.5,
patience=20,
embeddings_storage_mode='gpu',
shuffle=True,

**Comment by stefan-it** [comment]
Hi @krzysztoffsiuwa ,

try to use all layers from the XLNet model in combination with [scalar mix](https://github.com/zalandoresearch/flair/blob/master/resources/docs/embeddings/TRANSFORMER_EMBEDDINGS.md#scalar-mix).

By default only the first layer from the XLNet model is used. Just use:

```python

embeddings = XLNetEmbeddings(pretrained_model_name_or_path="xlnet-large-cased", layers="0,1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24", use_scalar_mix=True)
```

**Comment by krzysztoffsiuwa** [comment]
Maybe one more question: if i have an already trained XLNet model without the scalar mix option, so right now it output only embeddings from one layer, can i use that model and ask it to output embeddings from all the layers?

**Comment by krzysztoffiok** [comment]
Thank you again for response. I'm looking into my test runs and i see that in fact i omitted "0" layer, but only in some trials. Clumsy of me!

**Comment by krzysztoffiok** [comment]
I promised to publish some benchmarks i've made, so there they are <a href="https://github.com/krzysztoffiok/MPD-dataset">speed-quality tradeoff benchmarks</a>. They should be published in a research journal, but you know how peer-review article work...

It was carried out with flair 0.44 so BERT-related training times are already obsolete.

**Comment by alanakbik** [comment]
@krzysztoffiok thanks for sharing these experiments - very interesting results!

**Comment by krzysztoffiok** [comment]
Hi again @alanakbik,

i just got these experiments finally published. https://www.mdpi.com/2076-3417/10/10/3386 Thanks again all help.

**Comment by alanakbik** [comment]
Congrats! Thanks for sharing your results!

**Comment by alanakbik** [comment]
(Closing but feel free to reopen for more discussion)
